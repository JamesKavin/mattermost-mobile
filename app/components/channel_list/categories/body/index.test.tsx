// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithIntlAndTheme} from '@test/intl-test-helper';

import CategoryBody from './index';

const channels: TempoChannel[] = [
    {id: '1', name: 'Just a channel'},
    {id: '2', name: 'Highlighted!!!', highlight: true},
];

test('Category Body Component should match snapshot', () => {
    const {toJSON} = renderWithIntlAndTheme(
        <CategoryBody channels={channels}/>,
    );

    expect(toJSON()).toMatchSnapshot();
});
