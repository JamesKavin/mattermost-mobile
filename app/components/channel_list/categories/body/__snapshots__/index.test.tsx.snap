// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Body Component should match snapshot 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": "1",
        "name": "Just a channel",
      },
      Object {
        "highlight": true,
        "id": "2",
        "name": "Highlighted!!!",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingVertical": 4,
          }
        }
      >
        <Icon
          name="globe"
          style={
            Object {
              "color": "rgba(255,255,255,0.72)",
              "fontSize": 24,
              "lineHeight": 28,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "OpenSans",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Object {
                "color": "rgba(255,255,255,0.72)",
                "marginTop": 1,
                "paddingLeft": 12,
              },
              undefined,
            ]
          }
        >
          Just a channel
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingVertical": 4,
          }
        }
      >
        <Icon
          name="globe"
          style={
            Object {
              "color": "rgba(255,255,255,0.72)",
              "fontSize": 24,
              "lineHeight": 28,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "lineHeight": 24,
              },
              Object {
                "color": "rgba(255,255,255,0.72)",
                "marginTop": 1,
                "paddingLeft": 12,
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          Highlighted!!!
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
